internal/component/loki/write/write.go

```diff
@@ -228,10 +229,17 @@ func (c *Component) Update(args component.Arguments) error {
 
 func newEntryHandler(handler loki.EntryHandler, externalLabels model.LabelSet) loki.EntryHandler {
 	return loki.NewEntryMutatorHandler(handler, func(e loki.Entry) loki.Entry {
-		if len(externalLabels) == 0 {
-			return e
+
+		if len(externalLabels) > 0 {
+			e.Labels = externalLabels.Merge(e.Labels)
 		}
-		e.Labels = externalLabels.Merge(e.Labels)
+
+		for k := range e.Labels {
+			if strings.HasPrefix(string(k), "__") {
+				delete(e.Labels, k)
+			}
+		}
```

Comment:
Pretty sure it's not safe to modify `e.Labels` directly, there will be problems when you pass to multiple different clients. I think we have a test for that so I will trigger the ci to verify it.

Pretty sure we would need to check if we have any internal labels and only then perform a clone without these labels